util/headerreader/header_reader.go

```diff
@@ -27,7 +27,20 @@ import (
 )
 
 // A regexp matching "execution reverted" errors returned from the parent chain RPC.
-var ExecutionRevertedRegexp = regexp.MustCompile(`(?i)execution reverted|VM execution error\.?`)
+var executionRevertedRegexp = regexp.MustCompile(`(?i)execution reverted|VM execution error\.?`)
+
+// IsExecutionReverted returns true if the error is an "execution reverted" error or if the error is a rpc.Error with ErrorCode 3.
+func IsExecutionReverted(err error) bool {
+	if executionRevertedRegexp.MatchString(err.Error()) {
+		return true
+	}
+	var rpcError rpc.Error
+	ok := errors.As(err, &rpcError)
+	if ok && rpcError.ErrorCode() == 3 {
```

Comment:
I know that this was the task, and remember Lee finding that error code, but so far I found no confirmation that error code 3 does mean "execution reverted". 
I think remove this part or find other confirmation for it.
I do like the creation of IsExecutionReverted error function, and would be happy to merge it. Not critical.

Reply:
Good catch, so i looked around and it looks like the code is 6 based on https://github.com/OffchainLabs/go-ethereum/blob/master/core/vm/errors.go#L131, let me know if I should change it to 6 or it still makes sense to just just remove the code part.

Reply:
Actually nvm , here's the part where code 3 comes from - https://github.com/OffchainLabs/go-ethereum/blob/master/internal/ethapi/errors.go#L37

Reply:
Yeah. I think this looks right.